desktop/apps/editorial_features/test/routes.coffee

```diff
@@ -47,6 +47,42 @@ describe 'EOY route', ->
       @res.render.args[0][1].superSubArticles.length.should.equal 2
       done()
 
+describe 'Gucci route', ->
+  beforeEach ->
+    sinon.stub Backbone, 'sync'
+    Backbone.sync
+      .onCall 0
+      .yieldsTo 'success', { name: 'Artists For Gender Equality', sections: [{title: 'I. Present'}]}
+    @res = { render: sinon.stub(), locals: { sd: {} }, redirect: sinon.stub() }
+    @next = sinon.stub()
+    routes.__set__ 'sd', {EF_GUCCI: '123'}
+
+  afterEach (done) ->
+    Backbone.sync.restore()
+    done()
+
+  it 'sets a curation', (done) ->
+    @req = { params: { slug: 'past' } }
+    routes.gucci(@req, @res, @next)
+    _.defer => _.defer =>
```

Comment:
Because routes are often returning a promise implicitly (e.g. `curation.fetch(..)` returns a promise + coffeescript's expressions), we can avoid this `_.defer => _.defer =>` hack and instead do

````coffeescript
it 'sets a curation', ->
  # ...
  routes.gucci(@req, @res, @next).then =>
    @res.render.args ...
    # done() [1]
````

[1] In-fact even mocha supports promises now so `done` isn't actually necessary there as `routes.gucci().then =>` is returning a promise as well.

Not going to block this PR for it though as an ideal refactor would go further and change this whole file to ES6 + async/await. Just something to keep in mind.

Reply:
Yeah, no more CoffeeScript 😄 Force runs ES6 just fine these days.